### PR TITLE
feat: example with fraud detection

### PR DESCRIPTION
**Pull request by kboroszko**
Example on how to use pytorch_bigtable.

We use a credit card fraud detection dataset borrowed from [here](https://fraud-detection-handbook.github.io/fraud-detection-handbook/Chapter_3_GettingStarted/SimulatedDataset.html). It is neat because it can be dynamically generated to be as big as you wish, so maybe later if we want to stress test our setup, we will prepare a bigger dataset. For now it is using a pregenerated dataset.

We upload the training data into bigtable. The code for that is in `seed_bigtable.py`.

Then we use a simple, fully connected network to detect transactions that are fraudulent. The network can be trained using bigtable or data stored in a csv. 

We evaluate the model before and after training to verify that the network trains properly.

### CODE REVIEW

**Review by dopiera — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unoperate/pytorch-cbt/25)* status: 0 of 5 files reviewed, 6 unresolved discussions (waiting on @dopiera and @kboroszko)

___
*[plugin/example/fraud_example.py, line 1 at r1](https://reviewable.io/reviews/unoperate/pytorch-cbt/25#-MkhMLFv6_PNoK-VNdTB:-MkhMLFv6_PNoK-VNdTC:b-kla62i) ([raw file](https://github.com/unoperate/pytorch-cbt/blob/5f0cbdb0a041380720d48e2c83b2d1ade06d73a6/plugin/example/fraud_example.py#L1)):*
> ```Python
> # Copyright 2021 Google LLC
> ```

Please add a description of what this file does, what it illustrates, etc.
___
*[plugin/example/fraud_example.py, line 127 at r1](https://reviewable.io/reviews/unoperate/pytorch-cbt/25#-MkhJxWv8upxdkTtWWpk:-MkhJxWv8upxdkTtWWpl:b-ra6w36) ([raw file](https://github.com/unoperate/pytorch-cbt/blob/5f0cbdb0a041380720d48e2c83b2d1ade06d73a6/plugin/example/fraud_example.py#L127)):*
> ```Python
> 
>   parser = argparse.ArgumentParser("fraud_example.py")
>   parser.add_argument("-b", "--use_bigtable",
> ```

I think it will be better if you allow the user to specify the endpoint or the CSV filename rather than assuming the address/path.

Similarly, if there are output files, they should be configurable on the command line too.
___
*[plugin/example/seed_bigtable.py, line 1 at r1](https://reviewable.io/reviews/unoperate/pytorch-cbt/25#-MkhLyG-8Ai7LEi6t98K:-MkhLyG-8Ai7LEi6t98L:bj1g6es) ([raw file](https://github.com/unoperate/pytorch-cbt/blob/5f0cbdb0a041380720d48e2c83b2d1ade06d73a6/plugin/example/seed_bigtable.py#L1)):*
> ```Python
> # Copyright 2021 Google LLC
> ```

Please add a comment what this file does. Please also mention that it is for demonstrational purposes only and it is terribly inefficient, reads all data into RAM, etc. If the user wants to store more data, they should write their own tool to do so.
___
*[plugin/example/seed_bigtable.py, line 45 at r1](https://reviewable.io/reviews/unoperate/pytorch-cbt/25#-MkhKjZU5AhGeJoCb7zL:-MkhKjZU5AhGeJoCb7zM:bao1yrh) ([raw file](https://github.com/unoperate/pytorch-cbt/blob/5f0cbdb0a041380720d48e2c83b2d1ade06d73a6/plugin/example/seed_bigtable.py#L45)):*
> ```Python
> 
> 
> def read_data(start: [str, datetime] = "2018-04-01",
> ```

Why those defaults?
___
*[plugin/example/seed_bigtable.py, line 138 at r1](https://reviewable.io/reviews/unoperate/pytorch-cbt/25#-MkhLJzL8xB2ORVSsKZS:-MkhLJzL8xB2ORVSsKZT:b-tsxwcj) ([raw file](https://github.com/unoperate/pytorch-cbt/blob/5f0cbdb0a041380720d48e2c83b2d1ade06d73a6/plugin/example/seed_bigtable.py#L138)):*
> ```Python
> 
>   print("Seed bigtable")
>   os.environ["BIGTABLE_EMULATOR_HOST"] = "172.17.0.1:8086"
> ```

Please make it configurable as you did in the other file. Same for the CSV.
___
*[plugin/pytorch_bigtable/bigtable_dataset.py, line 105 at r1](https://reviewable.io/reviews/unoperate/pytorch-cbt/25#-MkhKPXx0FX5Fc5ASa81:-MkhKPXx0FX5Fc5ASa82:b-lmtji8) ([raw file](https://github.com/unoperate/pytorch-cbt/blob/5f0cbdb0a041380720d48e2c83b2d1ade06d73a6/plugin/pytorch_bigtable/bigtable_dataset.py#L105)):*
> ```Python
>     """
> 
>     if len(columns) != tensor.shape[1]:
> ```

It looks like we had a bug. Should we write a regression test?

**Review by kboroszko — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unoperate/pytorch-cbt/25)* status: 0 of 5 files reviewed, 18 unresolved discussions (waiting on @dopiera and @kboroszko)

___
*[plugin/example/fraud_example.py, line 127 at r1](https://reviewable.io/reviews/unoperate/pytorch-cbt/25#-MkhJxWv8upxdkTtWWpk:-MmMObKj16bRwMEFHMGI:b1g6212) ([raw file](https://github.com/unoperate/pytorch-cbt/blob/5f0cbdb0a041380720d48e2c83b2d1ade06d73a6/plugin/example/fraud_example.py#L127)):*
<details><summary><i>Previously, dopiera (Marek Dopiera) wrote…</i></summary><blockquote>

I think we should invest a bit more in the usability of this example. I'm guessing that for pyTorch <-> BT integration this will be the first thing the people will try - it should be as frictionless as possible.

Correct me if I'm wrong, but here's my understanding: the train set can live either in Bigtable or in a CSV file, but the test set can only line in CSV.

How about we do the following:
* drop `--use_bigtable` flag
* rename `--train_set` to `--train_set_csv`
* rename `--test_set` to `--test_set_csv`
* introduce `--train_set_table` and `--test_set_table`
* expect that if the user specified any of the `_table` options, `--project_id` and `--instance_id` are set, too
* forbid setting both `_table` and `_csv` options for both train and test sets (use `add_mutually_exclusive_group`)
</blockquote></details>

We're slowly approaching a point at which specifying a path to a json with config will be more feasible...
___
*[plugin/example/fraud_example.py, line 158 at r2](https://reviewable.io/reviews/unoperate/pytorch-cbt/25#-MmIg54A5TjBxymjaOLa:-MmMOpeLEnPr-YxbbIBD:b-896fix) ([raw file](https://github.com/unoperate/pytorch-cbt/blob/bf2eb1de5fb88e532c514c2f88eff52c6a169d48/plugin/example/fraud_example.py#L158)):*
<details><summary><i>Previously, dopiera (Marek Dopiera) wrote…</i></summary><blockquote>

> ```
> emulator_host
> ```
Usually, `addr` is `host:port`, so this should be `emulator_addr` (I know - the envvar is called `BIGTABLE_EMULATOR_HOST`, but I think we shouldn't promulgate it). To be honest, `addr` is also a bit misleading - URI is the proper term here because addr is also the IP alone, but given that "address" is also commonly used to denote the whole thing you enter to your browser, I think `addr` is good enough - certainly better than `host`, which almost always denotes the IP address or the DNS name without any port.
</blockquote></details>

Done.
___
*[plugin/example/fraud_example.py, line 161 at r2](https://reviewable.io/reviews/unoperate/pytorch-cbt/25#-MmIm7dlESuhvubOzZok:-MmMgJxb1svaZwfCI2s4:b91dch0) ([raw file](https://github.com/unoperate/pytorch-cbt/blob/bf2eb1de5fb88e532c514c2f88eff52c6a169d48/plugin/example/fraud_example.py#L161)):*
<details><summary><i>Previously, dopiera (Marek Dopiera) wrote…</i></summary><blockquote>

> ```
> --train_set
> ```
Please indicate the format, e.g. "as produced by the ... script" or something along these lines. Similarly, for the `_table` version, please indicate how to populate that table.

Also, please use `argparse.FileType` for files - it will provide meaningful validation and support for '-' for stdin.
</blockquote></details>

yeah, but we pass it later to the pandas.read_csv() so we don't want it to support '-'. We want it to be a valid path, nothing else. 
Pandas will throw a meaningful error if it fails to open the file but I added manual validation for clarity.
